src/main/java/frc/robot/classes/TunableNumber.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot.classes;
+
+import java.util.function.Consumer;
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Robot;
+
+public class TunableNumber extends SubsystemBase{
+    private ShuffleboardTab tuningTab = Shuffleboard.getTab("Tuning");
+    private GenericEntry numberEntry;
+    private String name;
+    private double value;
+    private Consumer<Double> consumer;
+
+    public TunableNumber(String name, double defaultValue, Consumer<Double> consumer){
+        this.name = name;
+        this.value = defaultValue;
+        this.consumer = consumer;
+        if (Robot.tuningMode){
+            numberEntry = tuningTab.add(name, defaultValue).getEntry();
+        }
+    }
+
+    public void periodic(){
+        if (Robot.tuningMode){
+            value = numberEntry.getDouble(0);
+            consumer.accept(value);
```

Comment:
If and only if the number has changed, I would like it to log that entry using high speed logging

Reply:
If we think that is too complex to hold up the PR, I am ok with a card in todo for this

Reply:
Made the number check and appended it to logging.